quodlibet/quodlibet/qltk/information.py

```diff
@@ -34,18 +36,49 @@ def Label(label=None):
     return l
 
 
+class TitleLabel(Gtk.Label):
+
+    def __init__(self, text, is_markup=False):
+        super(TitleLabel, self).__init__()
+        self.set_ellipsize(Pango.EllipsizeMode.END)
+        markup = text if is_markup else ("<i>%s</i>" % util.escape(text))
+        text = "<span size='xx-large'>%s</span>" % markup
+        self.set_markup(text)
+
+
 def Frame(name, widget):
+    def hx(value):
+        return hex(int(value * 255))[2:]
     f = Gtk.Frame()
-    f.set_shadow_type(Gtk.ShadowType.NONE)
+    cols = get_colors(f)
+    c = cols[0]
+    qltk.add_css(f, '* {opacity: 0.9}')
     l = Gtk.Label()
-    l.set_markup("<u><b>%s</b></u>" % name)
+    l.set_markup("<span color='#{0}{1}{2}{3}'>{4}</span>".format(
+        hx(c.red), hx(c.green), hx(c.blue), hx(c.alpha), name))
     f.set_label_widget(l)
-    a = Align(top=3, left=12)
+    a = Align(top=6, left=12, bottom=6, right=6)
     f.add(a)
     a.add(widget)
     return f
 
 
+def Table(rows):
+    t = Gtk.Table(n_rows=rows, n_columns=2)
+    t.set_col_spacings(6)
+    t.set_row_spacings(6)
+    t.set_homogeneous(False)
+    return t
+
+
+def get_colors(widget):
+    context = widget.get_style_context()
```

Comment:
Yeah, it's not obvious and only a problem since gtk 3.20+. The problem is that the getters invalidate things for gtk and when called from the wrong code path you get in an invalidation loop. (new-state -> update css/drawing -> new-state ...).